app/services/db/monthly_sp_auth_count/total_monthly_auth_counts_within_iaa_window.rb

```diff
@@ -13,11 +13,13 @@ module TotalMonthlyAuthCountsWithinIaaWindow
       # @param [String] iaa
       # @param [Date] iaa_start_date
       # @param [Date] iaa_end_date
+      # @param [Proc,#call] query_with_timeout block that sets up a timeout
       # @return [PG::Result,Array]
-      def call(issuer:, iaa:, iaa_start_date:, iaa_end_date:)
+      def call(issuer:, iaa:, iaa_start_date:, iaa_end_date:, &query_with_timeout)
         return [] if !iaa_start_date || !iaa_end_date
 
         iaa_range = (iaa_start_date..iaa_end_date)
+        query_with_timeout = method(:default_call) if !block_given?
```

Comment:
all this noise about block handling is because `#query_with_timeout` is an instance method on `BaseReport` so if I moved it to be a class method...that could all go away.

and the more I think about that, the more obvious that's the right choice.... will update

Reply:
yep, makes sense to me!

Reply:
updated in 18bf999a853aac759f90c69eac6423c6b8328dd9